api/base.py

```diff
@@ -39,30 +42,25 @@ def __init__(self, request=None, response=None):
                 log.debug('looked up cached token in %dms' % ((datetime.datetime.utcnow() - request_start).total_seconds() * 1000.))
             else:
                 r = requests.get(self.app.config['oauth2_id_endpoint'], headers={'Authorization': 'Bearer ' + access_token})
-                if r.status_code == 200:
+                if r.ok:
                     identity = json.loads(r.content)
                     self.uid = identity.get('email')
                     if not self.uid:
                         self.abort(400, 'OAuth2 token resolution did not return email address')
                     self.app.db.authtokens.replace_one({'_id': access_token}, {'uid': self.uid, 'timestamp': request_start}, upsert=True)
+                    self.app.db.users.update_one({'_id': self.uid, 'firstseen': {'$exists': False}}, {'$set': {'firstseen': request_start}})
                     log.debug('looked up remote token in %dms' % ((datetime.datetime.utcnow() - request_start).total_seconds() * 1000.))
 
-                    # Opportunistically set user's avatar based on their auth provider
-                    # TODO: after api starts reading toml config, switch on
-                    # auth.provider rather than manually comparing endpoint URL.
+                    # Set user's auth provider avatar
+                    # TODO: after api starts reading toml config, switch on auth.provider rather than manually comparing endpoint URL.
                     if self.app.config['oauth2_id_endpoint'] == 'https://www.googleapis.com/plus/v1/people/me/openIdConnect':
-                        avatar = identity.get('picture')
-                        # NOTE: Google URLs have a size attached. This code removes that parameter.
-                        # One could also set the size explicitly.
-                        # from urllib import urlencode
-                        # from urlparse import urlparse, urlunparse, parse_qs
-                        # u = urlparse(avatar)
-                        # query = parse_qs(u.query)
-                        # query.pop('sz', None)
-                        # u = u._replace(query=urlencode(query, True))
-                        # avatar = urlunparse(u)
-                        if avatar:
-                            r = self.app.db.users.update_one({'_id': self.uid, 'avatar': {'$ne': avatar}}, {'$set':{'avatar': avatar, 'modified': request_start}})
+                        provider_avatar = identity.get('picture')
+                        # Remove attached size param from URL.
+                        u = urlparse.urlparse(provider_avatar)
```

Comment:
You'll still want an `if provider_avatar` here to protect against crashing if the response does not contain an avatar.